modules/docs/lib/widgets/enhancedComponent.tsx

```diff
@@ -0,0 +1,199 @@
+
+const fileNameToCategoryMap: Record<string, string[]> = {
+  Layout: [
+    'utils/layout',
+    'utils/gridArea',
+    'utils/gridItem',
+    'utils/flexItem',
+    'utils/position',
+    'utils/space',
+    'utils/stack',
+  ],
+  Color: ['utils/color', 'utils/background'],
+  Border: ['utils/border'],
+  Depth: ['utils/depth'],
+  Text: ['utils/text'],
+  Other: ['utils/other'],
+};
+
+registerWidget<EnhancedComponentValue>('enhancedComponent', ({value, doc, meta}) => {
+  const groups = groupProps(value.props);
+  const parentComponentName = React.useContext(ParentComponentNameContext);
+  const parentComponentJSDoc = React.useContext(ParentComponentJSDocContext);
+  const hideHeader = meta && meta.hideHeader === true;
+
+  const intro = (
+    <>
+      {!parentComponentName && !hideHeader && value.displayName ? (
+        <Heading>{value.displayName}</Heading>
+      ) : null}
+
+      <ParentComponentNameContext.Provider value="">
+        <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+          <MdxJSToJSX>{parentComponentJSDoc.description || doc?.description || ''}</MdxJSToJSX>
+        </ParentComponentJSDocContext.Provider>
+      </ParentComponentNameContext.Provider>
+    </>
+  );
+
+  // We don't want to re-document a subcomponent that is a container component with a different
+  // model. If we detect another container, only render the intro and a link to the container
+  // component
+  if (parentComponentName && value.componentType === 'container') {
+    return (
+      <>
+        {intro}
+        <MDX as="p">
+          This component references the{' '}
+          <ParentComponentNameContext.Provider value="">
+            <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+              <SymbolDialog value={{kind: 'symbol', name: value.displayName || ''}} />
+            </ParentComponentJSDocContext.Provider>
+          </ParentComponentNameContext.Provider>{' '}
+          component.
+        </MDX>
+      </>
+    );
+  }
+
+  return (
+    <>
+      {intro}
+      {value.styleComponent ? (
+        <>
+          <Heading headingOffset={1}>Style Component</Heading>
+          <MDX as="p">
+            <code>
+              <ParentComponentJSDocContext.Provider value={defaultJSDoc}>
+                <Value value={value.styleComponent} />
+              </ParentComponentJSDocContext.Provider>
+            </code>
+          </MDX>
+        </>
+      ) : null}
+      <Heading headingOffset={1}>Props</Heading>
+      {value.baseElement && (
+        <MDX as="p">
+          Props extend from <Value value={value.baseElement} />. Changing the <code>as</code> prop
+          will change the element interface.
+        </MDX>
+      )}
+      {value.componentType === 'container' && value.model ? (
+        <MDX as="p">
+          Props extend from <SymbolDialog value={{kind: 'symbol', name: `${value.model}Config`}} />.
+          If a <code>model</code> is passed, model config is ignored.
```

Comment:
I had some trouble understanding this in the context of the docs.

Before:
<img width="895" alt="Screenshot 2023-02-22 at 5 28 08 PM" src="https://user-images.githubusercontent.com/1075861/220794916-4cdb893e-12b9-43e6-9258-0496a0b318ed.png">


After:

<img width="597" alt="Screenshot 2023-02-22 at 5 27 56 PM" src="https://user-images.githubusercontent.com/1075861/220794980-88346c32-121d-4240-b1a5-599dbcc7f89e.png">

I'm not even sure what to suggest, but what we had before made sense but what we have now is confusing to me.

Reply:
I'm not always good at explaining something tersely, so let me explain and see if there's a more terse explanation to come out of it.

Container components are responsible for creating and/or delegating a model. If a `model` is passed into a container component, that will be the `model` delegated to subcomponents. If a `model` is not passed, model config can be accepted directly as props. The model config props will be passed to a `modelHook` to create a `model` to delegate to subcomponents. There's a limitation in Typescript that prevents the types from regecting a `model` prop if any config props are present, so the types say all are optional. The implementation says the presents of a `model` prop takes presedence and any config props are ignored.

https://github.com/Workday/canvas-kit/blob/852643e/modules/docs/mdx/COMPOUND_COMPONENTS.mdx#container-components

We've thrown around the idea of a prop called `modelConfig` or `config` that is a holder for all model config rather than spread as props, but that would be a big breaking change.

Reply:
Also this text is autogenerated. If we update, it will change everywhere.

Reply:
How about: "Props extend from ActionBarModelConfig. If a `model` is passed, props from ActionBarModelConfig are ignored."

Reply:
I'll update to @jamesfan's suggestion